src/edu/duke/cs/osprey/confspace/AbstractTupleMatrix.java

```diff
@@ -522,4 +524,134 @@ public boolean equals(AbstractTupleMatrix<?> other) {
 
 		return true;
 	}
+
+	public String diff(int cellWidth, Function<T, String> formatter, EnergyMatrix scoreAnalysis) {
```

Comment:
Ideally, superclasses shouldn't be dependent on their subclasses. Is there a better place to put this 'diff' method? Maybe in 'EnergyMatrix' instead?